resolveImportsCached.mjs

```diff
@@ -1,7 +1,16 @@
+// @ts-check
+
 import resolveImports from "./resolveImports.mjs";
 
+/** @typedef {import("./resolveImports.mjs").ResolvedImports} ResolvedImports */
```

Comment:
Having to import types across modules get's messy, and there is a better way. Ultimately, what you intend to document is that it resolves whatever `resolveImports` resolves. You can do that without having to reach across modules and pull out types by doing `ReturnType<typeof resolveImports>`.

Once you do that, you realise actually, TypeScript would have inferred that same thing anyway, and manually specifying the return type is pointless. Generally it's a good idea to not hardcode the return type in function JSDoc, and let TS infer what it is itself. The downside is that you might not realise that the function you wrote returns something different than you had anticipated.